internal/tools/semconv-gen/generator.go

```diff
@@ -0,0 +1,319 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	flag "github.com/spf13/pflag"
+)
+
+func main() {
+	cfg := config{}
+	flag.StringVarP(&cfg.inputPath, "input", "i", "", "Path to semantic convention definition YAML")
+	flag.StringVarP(&cfg.outputPath, "output", "o", "semconv", "Path to output target. Must be either an absolute path or relative to the repository root.")
+	flag.StringVarP(&cfg.containerImage, "container", "c", "otel/semconvgen", "Container image ID")
+	flag.StringVarP(&cfg.outputFilename, "filename", "f", "", "Filename for templated output. If not specified 'basename(inputPath).go' will be used.")
+	flag.StringVarP(&cfg.templateFilename, "template", "t", "template.j2", "Template filename")
+	flag.Parse()
+
+	cfg, err := validateConfig(cfg)
+	if err != nil {
+		fmt.Println(err)
+		flag.Usage()
+		os.Exit(-1)
```

Comment:
nit: this is not POSIX compliant. TBH I think that have never seen negative exit codes 😉 
 https://tldp.org/LDP/abs/html/exitcodes.html

AFAIK `2` is generally used for bad input

```suggestion
		os.Exit(2)
```

Reply:
This is what `flag` package does: https://github.com/golang/go/blob/485474d204297ce9ff2b7f6c2a2b0e048c61fb8e/src/flag/flag.go#L317

Reply:
Shells have a slightly different range for valid exit codes, reserving 128-255.  The docs you link also specify that `2` is intended to represent "[m]isuse of shell builtins (according to Bash documentation)".  This isn't a shell utility though, so the relevant portion of the POSIX spec would be https://pubs.opengroup.org/onlinepubs/009695399/functions/exit.html.  This specifies that only the lower 8 bits of the exit code will be available, leading to an [exit code of 255](https://play.golang.org/p/xH8dT4uNg-z).  Maybe I'm old and times have changed out from under me, but `exit -1` is how I've always signaled abnormal termination.

Reply:
Resolve :-)